src/NDItk/multigroup/Structure/test/Structure.test.cpp

```diff
@@ -0,0 +1,191 @@
+// include Catch2
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+using Catch::Matchers::WithinRel;
+
+// what we are testing
+#include "NDItk/multigroup/Structure.hpp"
+
+// other includes
+
+// convenience typedefs
+using namespace njoy::NDItk;
+using Structure = multigroup::Structure;
+
+std::string chunk();
+void verifyChunk( const Structure& );
+std::string chunkWithInsufficientNumberBoundaries();
+std::string chunkWithNonDescendingBoundaries();
+std::string chunkWithNonUniqueBoundaries();
+
+SCENARIO( "Structure" ) {
+
+  GIVEN( "valid data for a Structure instance" ) {
+
+    std::string record = chunk();
+
+    WHEN( "the data is given explicitly" ) {
+
+      std::vector< double > boundaries = { 20., 18., 16., 14., 10., 5, 1, 1e-11 };
+
+      Structure chunk( std::move( boundaries ) );
+
+      THEN( "a Structure can be constructed and members can "
+            "be tested" ) {
+
+        verifyChunk( chunk );
+      } // THEN
+
+      THEN( "the record can be printed" ) {
+
+        std::string buffer;
+        auto output = std::back_inserter( buffer );
+        chunk.print( output );
+
+        CHECK( buffer == record );
+      } // THEN
+    } // WHEN
+
+    WHEN( "the data is read using iterators" ) {
+
+      auto iter = record.begin() + 8;
+      auto end = record.end();
+
+      Structure chunk;
+      chunk.read( iter, end, 8 );
+
+      THEN( "a Structure can be constructed and members can "
+            "be tested" ) {
+
+        verifyChunk( chunk );
+      } // THEN
+
+      THEN( "the record can be printed" ) {
+
+        std::string buffer;
+        auto output = std::back_inserter( buffer );
+        chunk.print( output );
+
+        CHECK( buffer == record );
+      } // THEN
+    } // WHEN
+  } // GIVEN
+
+  GIVEN( "invalid data for a Structure instance" ) {
+
+    WHEN( "the number of boundary values is insufficient" ) {
+
+      THEN( "an exception is thrown" ) {
+
+        std::vector< double > empty = {};
+        std::vector< double > one = { 1 };
+
+        CHECK_THROWS( Structure( std::move( empty ) ) );
+        CHECK_THROWS( Structure( std::move( one ) ) );
+      } // THEN
+    } // WHEN
+
+    WHEN( "the boundary values are not in descending order" ) {
+
+      THEN( "an exception is thrown" ) {
+
+        std::vector< double > wrong = { 1., 20. };
```

Comment:
Can you make this something like { 3., 1., 20. }?

Reply:
Or both.

Reply:
Done. Changed in fix/review